ci/Jenkinsfiles/cleanup-exports.groovy

```diff
@@ -0,0 +1,126 @@
+*
+* Cleans up oldest distributions marked as "next" on preview explorer site, keeping only a given number of them.
+* Allows avoiding piling up exports from nuxeo builds on preview.
+* Triggered daily by a cron.
+*/
+properties([
+  [$class: 'GithubProjectProperty', projectUrlStr: 'https://github.com/nuxeo/nuxeo-explorer/'],
+  [$class: 'BuildDiscarderProperty', strategy: [$class: 'LogRotator', daysToKeepStr: '60', numToKeepStr: '10', artifactNumToKeepStr: '5']],
+  disableConcurrentBuilds(),
+])
+
+pipeline {
+
+  agent {
+    label 'jenkins-nuxeo-package-11'
+  }
+
+  triggers {
+    // every day between 8AM and 10AM
+    cron('H H(8-10) * * *')
+  }
+
+  parameters {
+    string(name: 'NUMBER_KEEP', defaultValue: '15', description: 'Number of exports to keep.')
+    string(name: 'TARGET_URL', defaultValue: 'https://preview-nuxeo-explorer-master.platform.dev.nuxeo.com/nuxeo', description: 'Target Explorer instance to cleanup.')
+    string(name: 'TARGET_CREDS_ID', defaultValue: 'explorer-preview', description: 'Target Explorer instance credentials id.')
+  }
+
+  environment {
+    CURL_OPTIONS = "--fail --location --connect-timeout 180 --max-time 300 --retry 2"
+  }
+
+  stages {
+    stage('Check Parameters') {
+      steps {
+        script {
+          def errorMessage = ""
+          if (params.TARGET_URL.isEmpty() || params.TARGET_CREDS_ID.isEmpty()) {
+            errorMessage = 'TARGET_URL and TARGET_CREDS_ID parameters are mandatory'
+          } else if (!params.NUMBER_KEEP.isInteger() || params.NUMBER_KEEP.toInteger() < 0) {
+            errorMessage = "Invalid number of exports '${params.NUMBER_KEEP}' (expecting integer >= 0)"
+          }
+          if (!errorMessage.isEmpty()) {
+            currentBuild.result = 'ABORTED';
+            currentBuild.description = "${errorMessage}"
+            echo "${errorMessage}"
+            error(currentBuild.description)
+          }
+        }
+      }
+    }
+    stage('Set Labels') {
+      steps {
+        container('maven') {
+          echo """
+          ----------------------------------------
+          Set Kubernetes resource labels
+          ----------------------------------------
+          """
+          echo "Set label 'branch: export' on pod ${NODE_NAME}"
+          sh """
+            kubectl label pods ${NODE_NAME} branch=export
+          """
+        }
+      }
+    }
+    stage('Cleanup exports') {
+      steps {
+        container('maven') {
+          script {
+            echo """
+            ----------------------------------------
+            Cleanup Exports on ${TARGET_URL}
+            ----------------------------------------"""
+            withCredentials([usernameColonPassword(credentialsId: TARGET_CREDS_ID, variable: 'EXPLORER_PASS')]) {
+              def curlCommand = "curl --user ${EXPLORER_PASS} ${CURL_OPTIONS}"
+              def query = "SELECT * FROM NXDistribution WHERE nxdistribution:aliases=next ORDER BY nxdistribution:version"
```

Comment:
Note version ordering here may take 11.2.10 as more recent than 11.2.3 --> maybe order on release date instead?

Reply:
+ @efge tells me it should be "nxdistribution:aliases='next'" (with quotes), so it's weird that the query is executed correctly ( i tested "for real") -> to be checked

Reply:
For info it goes through conversion to page providers to ES --> seems it's more lenient (but adding the quotes is better, it works too)

Reply:
changes done